### PR TITLE
Junit test

### PR DESCRIPTION
**Pull request by carolinaWolox**
## Summary
- Junit test
## Trello Card
https://trello.com/c/uKZUWHek/28-junit-testing

### CODE REVIEW

File: src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        String bookPath = "comedyTest";
+        Optional<Book> optionalBook = Optional.of(book);
+
+        when(bookRepository.findByTitle(bookPath)).thenReturn(optionalBook);
+
+        mvc.perform(get("/books/title/{bookTitle}", "comedyTest"))
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title").value("comedyTest"));
+
+        //Assertions.assertEquals("application/json", result.getResponse().getContentType());
```
Comment (juanwongramirez) [inline src/test/java/wolox/training/controllers/BookControllerTest.java]: Please avoid to leave commented code

File: src/test/java/wolox/training/persistence/UserRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import wolox.training.models.UserBook;
+import wolox.training.repositories.UserRepository;
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class UserRepositoryIntegrationTest {
```
Comment (juanwongramirez) [inline src/test/java/wolox/training/persistence/UserRepositoryIntegrationTest.java]: Please change the class name, in this class you try to test repository methods and not integration with other components

**Review by juanwongramirez — CHANGES_REQUESTED** [review]

File: src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                .andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void whenUpdateBook_ThenUpdatedOk() throws Exception {
+
+        book.setId(1);
+        Optional<Book> optionalBook = Optional.of(book);
+        String bookBody = "{\"id\":\"1\",\"genre\":\"drama\",\"author\":\"author\",\"image\":\"image\",\"title\":\"book3\",\"subtitle\":\"subtitle\",\"publisher\":\"publisher\",\"year\":\"2020\",\"pages\":\"pages\",\"isbn\":\"isbn\"}";
```
Comment (julianvasquez-wolox) [inline src/test/java/wolox/training/controllers/BookControllerTest.java]: Hello! It's ok, but it's easier to use the ObjectMapper.writeValueAsString(object) to build a JsonString. 😁

Comment (carolinaWolox) [reply src/test/java/wolox/training/controllers/BookControllerTest.java]: it was resolved at last commit

**Review by juanwongramirez — APPROVED** [review]

**Review by julianvasquez-wolox — APPROVED** [review]

**Pull request merged**